Library/Homebrew/extend/os/mac/diagnostic.rb

```diff
@@ -32,6 +33,22 @@ def check_for_unsupported_osx
         EOS
       end
 
+      def check_for_prerelease_xcodes
+        return if ARGV.homebrew_developer?
+        # Running pre-release Xcodes on a pre-release OS is expected
+        # and likely to cause less problems than stable Xcodes.
+        return if OS::Mac.prerelease?
+        return unless MacOS::Xcode.installed?
+        return unless MacOS::Xcode.prerelease?
+
+        <<-EOS.undent
+          You are using a pre-release version of Xcode.
+          You may encounter build failures or other breakages.
+          Please consider creating pull-requests instead of
+          filing issues.
```

Comment:
I don't think we need to soften this. We're already saying “please” and it's not like we're expecting something that is unheard of, i.e. “if you're using beta software, you understand that things *will* be broken and people might not be able to help you”. No?